cloud/amazon/cloudfront.py

```diff
@@ -0,0 +1,557 @@
+        streaming_distribution
+    required: True
+    choices: [
+            'distribution',
+            'origin_access_id',
+            'invalidation',
+            'streaming',
+            ]
+    default: None
+  policy:
+    description:
+      - The path to the properly json formatted policy file or a
+        properly formatted json policy as string, see
+        https://github.com/ansible/ansible/issues/7005#issuecomment-42894813
+        on how to use it properly
+      - Used for creation and updating of resources
+    required: false
+    default: None
+  resource_id:
+    description:
+      - Required when creating an invalidation against a distribution
+      - Required for removal of distributions and origin_access_ids
+    required: false
+  state:
+    description:
+      - present to ensure resource is created or updated.
+      - absent to remove resource
+    required: false
+    default: present
+    choices: [ "present", "absent"]
+  wait_for_deployed:
+    description:
+      - distributions and streaming_distributions need to be disabled
+        before you can remove them.
+      - Setting this to yes will allow this module to disable the
+        distribution on your behalf, wait
+      - until the status has changed to "Deployed" before removing your
+        distribution. This has a timeout of 15 mins which is
+        the recommended value from AWS.
+      - this setting can also be used ensure a distribution,
+        invalidation, origin_access_id is created or updated
+    required: false
+    choices: ['yes', 'no']
+    default: no
+author: Karen Cheng(@Etherdaemon)
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Simple example of creating a new distribution with a json file
+- name: Create a new distribution
+  cloudfront:
+    type: distribution
+    state: present
+    policy: "{{ role_path }}/files/distribution.json"
+
+- name: Create a new distribution and wait for deployed
+  cloudfront:
+    type: distribution
+    state: present
+    wait_for_deployed: yes
+    policy: "{{ role_path }}/files/distribution.json"
+
+- name: Create a new origin access identity
+  cloudfront:
+    type: origin_access_id
+    state: present
+    policy: "{{ role_path }}/files/origin_access.json"
+
+#Disable and delete distribution
+- name: Disable and wait for status to change to "Deployed" and delete
+    distribution
+  cloudfront:
+    type: distribution
+    state: absent
+    resource_id: EEFF123DDFF
+    wait_for_deployed: yes
+
+- name: Create a new invalidation using policy template
+  cloudfront:
+    type: invalidation
+    resource_id: EEFF123DDFF
+    state: present
+    policy: " {{ lookup( 'template', 'invalidation.json.j2') }} "
+'''
+
+RETURN = '''
+result:
+  description: The result of the create, delete or update action.
+  returned: success
+  type: dictionary or a list of dictionaries
+'''
+
+try:
+    import json
+    import botocore
```

Comment:
you don't use `botocore` so I don't think it's worth checking the import, `boto3` will still raise an `ImportError` if it's missing since it goes on to import `botocore` internally